SQF/dayz_code/compile/dze_buildChecks.sqf

```diff
@@ -79,12 +79,15 @@ if (((count DZE_SafeZoneNoBuildItems) > 0) && {_classname in DZE_SafeZoneNoBuild
 	{
 		if ((player distance (_x select 0)) <  DZE_SafeZoneNoBuildDistance) exitWith { _canBuild = false; };
 	} forEach DZE_safeZonePosArray;
-	if !(_canBuild) exitWith { dayz_actionInProgress = false; format [localize "STR_EPOCH_PLAYER_166",_text,DZE_SafeZoneNoBuildDistance] call dayz_rollingMessages; [false, _isPole]; };
 };
 
-if (((count DZE_SafeZoneNoBuildNear) > 0) && {count (nearestObjects [_pos, DZE_SafeZoneNoBuildNear, DZE_BlacklistNoBuildDistance]) > 0}) exitWith {
+if !(_canBuild) exitWith { dayz_actionInProgress = false; format [localize "STR_EPOCH_PLAYER_166",_text,DZE_SafeZoneNoBuildDistance] call dayz_rollingMessages; [false, _isPole]; };
+
+_near = (nearestObjects [_pos,DZE_NoBuildNear,DZE_NoBuildNearDistance]);
```

Comment:
Move the search inside this check:
```sqf
if (count DZE_NoBuildNear > 0) then {
// do nearestObjects search
};
```
A nearestObjects search with an empty array is very taxing, especially greater than 50m.